uktrade/data-workspace
# Add published version to reference datasets

This change allows unpublished datasets to consolidate changes in to one major or minor update.

It does this by using a new field called published_version which whilst published mirrors that of version.
Upon being unpublished and changes made the numbers diverge (version number never being surfaced in the admin to avoid confusion). On republishing the numbers are checked and the next major or minor version is then set to both variables.

To test: 
 - Create a reference dataset 
 - Make changes add fields add data
 - Check the version number is yet to be set
 - Publish
 - Check version number now 1.0
 - Make changes
 - Check the version number follows current behaviour
 - Unpublish
 - Make multiple updates
 - Check version number does not change
 - Publish
 - Check version number is incremented only once (either major or minor depending on the changes)

Comments:
 - Unsure if the html templates should be updated to use published_version property rather than the version
 - Unsure if the approach taken to what data is shown in the admin is correct. Should it show version also so the user is aware changes have been made or is displaying both confusing.
- Migration sets defaults on fields which can cause db locking on large tables. This table has a hand full of rows but should it be split up anyway?




<img width="644" alt="Screenshot 2019-11-26 at 11 30 44" src="https://user-images.githubusercontent.com/8222658/69630985-5995d180-1045-11ea-9959-b882edcb4bce.png">
